src/utils/git.rs

```diff
@@ -267,21 +267,31 @@ pub fn fetch(
     })
 }
 
+fn get_ref_from_name<'repo>(repo: &'repo Repository, name: &str) -> Option<Reference<'repo>> {
+    match repo.find_branch(name, BranchType::Local) {
+        Ok(branch) => Some(branch.into_reference()),
+        Err(e) => { // toDo may be other errors here...
+            let tag_ref = repo.find_reference(&format!("refs/tags/{}", name)).expect("reference not found");
```

Comment:
`panic`ing when the reference definitely isn't what we should do, especially considering this can be used as a library.

Reply:
638721c